syntax = "proto3";

service Booking {
    rpc GetAllBookings(EmptyBooking) returns (stream BookingData) {}
    rpc GetBookingByUser(UserID) returns (BookingData) {}l
    rpc PostBookingByUser(NewBooking) returns (BookingData) {}
}

message UserID {
    string id = 1;
}

message BookingData {
    string userid = 1;
    repeated Date dates = 2;
}

message Date {
    string date = 1;
    repeated string movies = 2;
}

message NewBooking {
    string userid = 1;
    string date = 2;
    string movieid = 3;
}

message EmptyBooking {}